### PR TITLE
Refactor ci build scripts by using configuration generated by maven

### PR DESCRIPTION
**Pull request by cassand**
This PR remove configurations hard-coded into bash script and use maven properties instead.

A bash library to execute maven command with a consistent environment and to retrieve maven configuration is introduced in scripts/utils/maven.sh.

### CODE REVIEW

File: scripts/ci/build.sh [hunk 1]
```diff
@@ -18,30 +18,27 @@ set -e
...
-  source $dir/gcloud-init.sh
+  source ${dir}/gcloud-init.sh
 fi
 
-if [ -n "$DOCKER_NAMESPACE" ]; then
-  PROJECT_NAMESPACE="$DOCKER_NAMESPACE"
-else
-  PROJECT_NAMESPACE="gcr.io/$(gcloud info \
+pushd ${projectRoot}
+  export DOCKER_TAG_LONG=$(git rev-parse --short HEAD)
```
Comment (aslo) [inline scripts/ci/build.sh]: Do you need to be at the git repository root to execute this command? I would have thought you can run this from anywhere within the repository.

File: scripts/build.sh [hunk 2]
```diff
@@ -21,12 +21,10 @@
...
 
-if [ -z "$DOCKER_NAMESPACE" ]; then
-  set -- $@ "-Ddocker.project.namespace=$DOCKER_NAMESPACE"
-fi
+source ${projectRoot}/scripts/utils/maven.sh
 
-pushd $projectRoot
-mvn -P-local-docker-build -P-test.local "$@" clean install
+pushd ${projectRoot}
+  maven_utils::execute $@ clean install
```
Comment (aslo) [inline scripts/build.sh]: I don't think it's necessary to pass all arbitrary arguments to this script into maven like this - I actually think it adds more potential to confuse people. I'd prefer if this script only passed in the maven arguments/properties that are necessary.

**Review by aslo — CHANGES_REQUESTED** [review]

File: scripts/build.sh (hunk 2 above)
Comment (cassand) [reply scripts/build.sh]: Indeed, `maven_utils::execute` can be used to pass arbitrary arguments so this is not needed anymore

**Comment by meltsufin** [comment]
@aslo and @cassand I'm not sure how we got here, but all of these scripts are getting really confusing for me. I'm having to think real hard every time I look at this to understand what's used where and why. Maybe renaming the scripts and adding more comments would help. I don't know. I'm open to ideas to make this better.

**Comment by aslo** [comment]
@meltsufin Agreed - we should probably be a bit more principled in what goes into these scripts. Let's chat about it offline.

**Review by meltsufin — APPROVED** [review]

**Review by aslo — APPROVED** [review]

**Pull request merged**